src/js/pages/Arbitration/index.jsx

```diff
@@ -36,20 +40,58 @@ class Arbitration extends Component {
     props.loadArbitration(props.escrowId);
   }
 
+  state = {
+    displayUsers: false,
+    selectedUser: null,
+    displayDialog: false
+  }
+
   componentDidMount() {
   }
 
+  openSolveDisputeDialog = () => {
+    this.setState({displayUsers: true});
+  }
+
+  selectUser = selectedUser => () => {
+    this.setState({selectedUser});
+  }
+
+  handleClose = () => {
+    this.setState({
+      displayUsers: false,
+      selectedUser: null,
+      displayDialog: false
+    });
+  }
+
+  displayDialog = show => (e) => {
+    if(e) e.preventDefault();
+
+    this.setState({displayDialog: show});
+    return false;
+  };
+
+  resolveDispute = () => {
+    this.setState({displayDialog: false, display: false});
+
+    this.props.resolveDispute(this.props.escrow.escrowId, this.state.selectedUser === this.props.escrow.buyer ? ARBITRATION_SOLVED_BUYER : ARBITRATION_SOLVED_SELLER )
+  }
+
   render() {
-    const {escrow, address} = this.props;
+    const {escrow, address, loading} = this.props;
+    const {displayUsers, selectedUser} = this.state;
 
     if(!escrow){
       return <Loading/>;
     }
 
+    if(loading) return <Loading mining={true} />;
+
     if(escrow.buyer === address || escrow.seller === address) return <ErrorInformation message="You cannot arbitrate your own disputes"/>;
+    if(escrow.arbitrator !== address) return <ErrorInformation message="You are not the arbitrator of this dispute"/>;
```

Comment:
It's better to put the errors (both of them, about the loadings, to avoid infinite loadings

Reply:
Above the other loading too 🤷‍♂

Reply:
ah but does it make sense? the other loading happens only while `escrow == null`  and these errors depend on the escrow not being null.

Reply:
Oops, you are totally right, sorry